ckanext/switzerland/helpers/backend_helpers.py

```diff
@@ -190,3 +183,58 @@ def ogdch_resource_display_name(res):
                 logic.NotAuthorized, AttributeError):
             return ""
     return resource_display_name
+
+
+def dataset_display_name(package_or_package_dict):
+    """
+    monkey patched version of ckan.lib.helpers.dataset_display_name which
+    extracts the correct translation of the dataset name
+    """
+    name = dataset_display_name_orig(package_or_package_dict)
+    return get_localized_value_for_display(name)
+
+
+def organization_link(organization):
+    """
+    monkey patched version of ckan.lib.helpers.organization_link which extracts
+    the correct translation of the organization name
+    """
+    organization['title'] = get_localized_value_for_display(
+        organization['title'])
+    return organization_link_orig(organization)
+
+
+def group_link(group):
+    """
+    monkey patched version of ckan.lib.helpers.group_link which extracts the
+    correct translation of the group title
+    """
+    url = url_for(controller='group', action='read', id=group['name'])
+    title = group['title']
+    try:
+        # the group creation message contains str(dict), so we must parse the
+        # string to fix it. If the title is just a string, a ValueError is
+        # thrown.
+        title = ast.literal_eval(title)
```

Comment:
is `title` here a dict as we have to do the `get_localized_value_for_display` action anyways? 
maybe we can name it that, to make it more clear.

Reply:
Hm, I tried to leave an explanatory comment here but I don't think I succeeded! It seems that, at this point, `title` can either be a string (e.g. `'Administration'`) or a dictionary represented by a string (e.g. `'{"de": "Administration", "fr": "Administration but in French"}'`). The latter case only happens when we are handling the group creation message, i.e. what is shown on the Activity Stream page, but not when handling group edit or delete messages.

If we get the dictionary representation, we have to do `ast.literal_eval` to transform it back to a dict before localizing. If it's a string, we try to eval it, but get a ValueError, which we ignore. Passing a string to `get_localized_value_for_display` does eventually return the same string, so it's okay to do that, but doesn't seem very efficient. I guess we can move that call into the try block as well. :thinking:

I will try to improve the comment with some of what I wrote here, if that's okay.

Reply:
I think I understood that part, that we have to do that guessing anyways. 
But I think it would just make it a little more clear to state, that this is not the final title, but rather a `title_thing`, `raw_title` (something that indicates in the name, that there is a title somewhere, but we don't know yet, in which form) It is very nitpicking, though, if you have an idea, go for it, but if not, I am fine with leaving it like that, too 🙂

Reply:
Thanks for explaining what you meant! I decided to keep the name `title` because it gets a bit complicated otherwise, with the many processing steps and the try/except. I hope this is all right for you.

I also added in the step here to decode accent characters, if necessary - of course this is only needed sometimes and not all the time :roll_eyes:

Reply:
That is fair, thank you for looking into this again :)